schemas/ships.json

```diff
@@ -0,0 +1,230 @@
+        "agility",
+        "hull",
+        "shields",
+        "actions",
+        "xws",
+        "maneuvers",
+        "maneuvers_energy",
+        "size"
+      ]
+    },
+    "non_huge": {
+      "type": "object",
+      "description": "Fields used by small and large ships only.",
+      "properties": {
+        "maneuvers": {
+          "type": "array",
+          "description": "The huge ship\\s maneuvers.",
+          "uniqueItems": false,
+          "maxItems": 6,
+          "minItems": 0,
+          "items": {
+            "type": "array",
+            "description": "Each element in this array contains a representation of the maneuvers available to the ship at a particular speed, determined its position in the array. ei. ship.maneuvers[1] will provide all maneuvers available to said ship at speed 1.\nThis array may be a short as required to provide accurate data, meaning that a missing speed 'index' indicates that the ship is is not capable of such speed.",
+            "uniqueItems": false,
+            "maxItems": 13,
+            "minItems": 0,
+            "items": {
+              "type": "integer",
+              "description": "This array is a representation of a ship' maneuvers at a particular speed.\n\nEach value on this array references a different maneuver depending on its index, which maps according to the following list:\n\t00 = Left Turn\n\t01 = Left Bank\n\t02 = Straight\n\t03 = Right Bank\n\t04 = Right Turn\n\t05 = Koiogran Turn\n\t06 = Segnor's Loop Left\n\t07 = Segnor's Loop Right\n\t08 = Tallon Roll Left\n\t09 = Tallon Roll Right\n\t10 = Backwards Left Bank\n\t11 = Backwards Straight\n\t12 = Backwards Right Bank\n\nPossible values in this array range from 0 to 3 and mean the following:\n\t0 = Maneuver unavailable\n\t1 = White maneuver\n\t2 = Green maneuver\n\t3 = Red maneuver\n\nThis array may be a short as required to provide accurate data, meaning that a missing value for a particular maneuver type indicates that said maneuver is not available to that particular ship at that particular speed.\n.",
+              "minimum": 0,
+              "maximum": 3
+            }
+          }
+        },
+        "size": {
+          "description": "The ship's size.",
+          "allOf": [
+            {
+              "description": "Ship size must be a valid size in size in the game.",
+              "$ref": "definitions.json#/definitions/size"
+            },
+            {
+              "type": "string",
+              "description": "This schema only applies to small or large ships.\nTherefore, ship size is restricted to small or large.",
+              "enum": [
+                "small",
+                "large"
+              ]
+            }
+          ]
+        }
+      },
+      "required": [
+        "name",
+        "faction",
+        "agility",
+        "attack",
+        "hull",
+        "shields",
+        "actions",
+        "xws",
+        "maneuvers",
+        "size"
+      ]
+    }
+  },
+  "anyOf": [
```

Comment:
A minor thing, but I guess this should be `oneOf`. At the moment a ship can never match both, but it's best to be strict about it.